modules/sfp_dehashed.py

```diff
@@ -0,0 +1,186 @@
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    # For a list of all events, check sfdb.py.
+    def watchedEvents(self):
+        return ["EMAILADDR"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["EMAILADDR_COMPROMISED", "PASSWORD_COMPROMISED",
+                "HASH_COMPROMISED", "RAW_RIR_DATA"]
+
+    # When querying third parties, it's best to have a dedicated function
+    # to do so and avoid putting it in handleEvent()
+    def query(self, qry):
+        headers = {
+            'Accept' : 'application/json',
+            'Authorization': "Basic " + base64.b64encode(self.opts['email'] + ":" + self.opts['api_key'])
+        }
+        res = self.sf.fetchUrl("https://api.dehashed.com/search?query=\"" + qry + "\"",
+                                headers=headers,
+                                timeout=15,
+                                useragent=self.opts['_useragent'])
+
+        if res['content'] is None:
+            self.sf.info("No Dehashed info found for " + qry)
+            return None
+
+        # Always always always process external data with try/except since we cannot
+        # trust the data is as intended.
+        try:
+            info = json.loads(res['content'])
+        except Exception as e:
+            self.sf.error("Error processing JSON response from Dehashed.", False)
+            return None
+
+        return info
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        # Once we are in this state, return immediately.
+        if self.errorState:
+            return None
+
+        # Log this before complaining about a missing API key so we know the
+        # event was received.
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Always check if the API key is set and complain if it isn't, then set
+        # self.errorState to avoid this being a continual complaint during the scan.
+        if self.opts['api_key'] == "" or self.opts['email'] == "":
+            self.sf.error("You enabled sfp_dehashed but did not set an email or API key!", False)
+            self.errorState = True
+            return None
+        
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+
+        self.results[eventData] = True
+
+        entries = list()
+        # To implement  : Number of pages 
+        # Fetch Dehashed data for incoming data (email)
+        jsonData = self.query(eventData)
+
+        if jsonData is None:
+            return None
+        
+        entries = jsonData.get('entries')
+        if entries is None:
+            return None
+
+        
+        evt.moduleDataSource = event.moduleDataSource
```

Comment:
This is not correct, since the data source is Dehashed, not the source of the email address event. The line should just be removed and leave it to the default behavior

Reply:
Okay, got it!